### PR TITLE
[Spark 4.0] Account for `CommandUtils.uncacheTableOrView` signature change.

### LINKED ISSUE #10710: [AUDIT] `uncacheTableOrView` changed in CommandUtils
Spark made a change in https://github.com/apache/spark/commit/6d41866e631, modifying the parameters to the `uncacheTableOrView`. We depend on this method and we should make the appropriate change.

### PR DESCRIPTION
**Pull request by mythrocks**
Fixes #10710.

This commit accounts for the changes in the signature of `CommandUtils.uncacheTableOrView` in Apache Spark 4.0. (See [SPARK-47191](https://github.com/apache/spark/pull/45289).)

### CODE REVIEW

**Comment by mythrocks** [comment]
Build

File: sql-plugin/src/main/spark311/scala/org/apache/spark/sql/hive/rapids/shims/CommandUtilsShimBase.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
```
Comment (razajafri) [inline sql-plugin/src/main/spark311/scala/org/apache/spark/sql/hive/rapids/shims/CommandUtilsShimBase.scala:1]: We don't need this trait but if you still want to keep it why not just keep it outside in the common source? Is it not applicable to a certain version of Spark?

Comment (mythrocks) [reply sql-plugin/src/main/spark311/scala/org/apache/spark/sql/hive/rapids/shims/CommandUtilsShimBase.scala:1]: Ok, that's doable.  This is the fallout from my reaction to the feedback in #10857. :)) I'll remove the base.

**Comment by mythrocks** [comment]
Build

**Review by razajafri — APPROVED** [review]

**Comment by mythrocks** [comment]
The CI run failed with the following error. (It looks unrelated to my current change.)
```
024-05-28T20:46:22.5133881Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/Spark321PlusDBShims.scala:44: Unused import
2024-05-28T20:46:22.5134484Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z] [ERROR] import org.apache.spark.sql.rapids.execution._
2024-05-28T20:46:22.5135036Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z] [ERROR]                                              ^
2024-05-28T20:46:22.5136444Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/spark330db/scala/org/apache/spark/rapids/execution/shims/GpuSubqueryBroadcastMeta.scala:97: type mismatch;
2024-05-28T20:46:22.5136788Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z]  found   : Int
2024-05-28T20:46:22.5137168Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z]  required: Seq[Int]
2024-05-28T20:46:22.5137918Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z] [ERROR]     GpuSubqueryBroadcastExec(s.name, s.index, s.buildKeys, broadcastBuilder())(
2024-05-28T20:46:22.5138437Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z] [ERROR]                                        ^
2024-05-28T20:46:22.5140243Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.458Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/spark330db/scala/org/apache/spark/rapids/execution/shims/GpuSubqueryBroadcastMeta.scala:40: private var broadcastBuilder in class GpuSubqueryBroadcastMeta is never used
2024-05-28T20:46:22.5140860Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.459Z] [ERROR]   private var broadcastBuilder: () => SparkPlan = _
2024-05-28T20:46:22.5141247Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.459Z] [ERROR]               ^
2024-05-28T20:46:22.5143138Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.459Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/spark330db/scala/org/apache/spark/rapids/execution/shims/GpuSubqueryBroadcastMeta.scala:102: private method getBroadcastModeKeyExprs in class GpuSubqueryBroadcastMeta is never used
2024-05-28T20:46:22.5143839Z [2024-05-28T20:45:49.822Z] [2024-05-28T20:45:38.459Z] [ERROR]   private def getBroadcastModeKeyExprs: Option[Seq[Expression]] = {
```

**Comment by mythrocks** [comment]
I've merged up for #10933.  I'll kick off the CI run shortly.

**Comment by mythrocks** [comment]
Build

**Comment by mythrocks** [comment]
This PR still fails on an unrelated error:
```
2024-05-29T17:47:06.1091539Z [2024-05-29T17:46:28.999Z] [2024-05-29T17:46:20.351Z] [ERROR] /home/ubuntu/spark-rapids/sql-plugin/src/main/spark330db/scala/org/apache/spark/rapids/execution/GpuSubqueryBroadcastMeta.scala:33: GpuSubqueryBroadcastMeta is already defined as class GpuSubqueryBroadcastMeta
2024-05-29T17:47:06.1092047Z [2024-05-29T17:46:28.999Z] [2024-05-29T17:46:20.352Z] [ERROR] class GpuSubqueryBroadcastMeta(
2024-05-29T17:47:06.1092396Z [2024-05-29T17:46:28.999Z] [2024-05-29T17:46:20.352Z] [ERROR]       ^
2024-05-29T17:47:06.1092787Z [2024-05-29T17:46:28.999Z] [2024-05-29T17:46:20.352Z] [ERROR] one error found
2024-05-29T17:47:06.1093385Z [2024-05-29T17:46:28.999Z] [2024-05-29T17:46:20.352Z] [INFO] ------------------------------------------------------------------------
```
#10945 should resolve it.

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

**Pull request merged**